src/plugins/discover/public/application/view_components/utils/use_index_pattern.ts

```diff
@@ -74,25 +74,25 @@ export const useIndexPattern = (services: DiscoverViewServices) => {
             handleIndexPattern();
           }
         } else {
-          try {
-            const ip = await fetchIndexPatternDetails(indexPatternIdFromState);
-            if (isMounted) {
-              setIndexPattern(ip);
-            }
-          } catch (error) {
-            if (isMounted) {
-              const warningMessage = i18n.translate('discover.indexPatternFetchErrorWarning', {
-                defaultMessage: 'Error fetching index pattern: {error}',
-                values: { error: (error as Error).message },
-              });
-              toastNotifications.addWarning(warningMessage);
-            }
+          const ip = await fetchIndexPatternDetails(indexPatternIdFromState);
+          if (isMounted) {
+            setIndexPattern(ip);
           }
         }
       }
     };
 
-    handleIndexPattern();
+    try {
+      handleIndexPattern();
```

Comment:
so instead of just handling the error case when enhancements is in not enabled. i wrapped the function in the same try catch. so in reality the delta change here was just moving where we put the try catch on